### PR TITLE
(maint) use golangci-lint in ci

### PR DESCRIPTION
**Pull request by tphoney**

### CODE REVIEW

**Comment by tphoney** [comment]
there are errors being hit in testing 

```
--- FAIL: TestGet (0.04s)
    --- FAIL: TestGet/Test_Get()_caching_errors (0.00s)
        source_test.go:166: all sources (1) failed
time="2024-08-08T09:32:13Z" level=info msg="NATS connecting" servers="nats://nats:4222,nats://localhost:4222" triesLeft=5
time="2024-08-08T09:32:13Z" level=info msg="NATS connected" ServerID=NB6BQL3ULIP3GLAKLIIHCZ4CTLAIMXMLCG5FGEMOVVG5OPSW5XBJCTSR URL:="nats://localhost:4222"
time="2024-08-08T09:32:13Z" level=info msg="NATS connecting" servers="nats://nats:4222,nats://localhost:4222" triesLeft=5
time="2024-08-08T09:32:13Z" level=info msg="NATS connected" ServerID=NB6BQL3ULIP3GLAKLIIHCZ4CTLAIMXMLCG5FGEMOVVG5OPSW5XBJCTSR URL:="nats://localhost:4222"
time="2024-08-08T09:32:13Z" level=info msg="NATS connecting" servers="nats://nats:4222,nats://localhost:4222" triesLeft=5
time="2024-08-08T09:32:13Z" level=info msg="NATS connected" ServerID=NB6BQL3ULIP3GLAKLIIHCZ4CTLAIMXMLCG5FGEMOVVG5OPSW5XBJCTSR URL:="nats://localhost:4222"
time="2024-08-08T09:32:14Z" level=warning msg="NATS not connected" inBytes=0 lastError="<nil>" outBytes=0 reconnects=1 status=RECONNECTING
--- FAIL: TestListSearchCaching (0.67s)
    --- FAIL: TestListSearchCaching/non-caching_of_OTHER_errors (0.00s)
        source_test.go:488: all sources (1) failed
    --- FAIL: TestListSearchCaching/non-caching_when_ignoreCache_is_specified (0.00s)
        source_test.go:514: all sources (1) failed
```

it looks like in the tests e.ExecuteQuerySync(context.Background(), &req) fails for the first time for some tests. This was previously hidden by not checking the error from the execute.  @DavidS-ovm thoughts ?

File: source_test.go [hunk 1]
```diff
@@ -488,29 +509,47 @@ func TestListSearchCaching(t *testing.T) {
...
+			t.Fatal(err)
+		}
+		_, _, err = e.ExecuteQuerySync(context.Background(), &q)
+		if err != nil {
+			t.Fatal(err)
+		}
 
 		if l := len(src.GetCalls); l != 2 {
-			t.Errorf("Exected 2 get calls, got %v", l)
+			t.Errorf("Executed 2 get calls, got %v", l)
```
Comment (DavidS-ovm) [inline source_test.go:544]:
```suggestion
			t.Errorf("Expected 2 get calls, got %v", l)
```

File: source_test.go [hunk 2]
```diff
@@ -488,29 +509,47 @@ func TestListSearchCaching(t *testing.T) {
...
 		}
 
 		if l := len(src.ListCalls); l != 2 {
-			t.Errorf("Exected 2 List calls, got %v", l)
+			t.Errorf("Executed 2 List calls, got %v", l)
```
Comment (DavidS-ovm) [inline source_test.go:548]:
```suggestion
			t.Errorf("Expected 2 List calls, got %v", l)
```

File: source_test.go [hunk 3]
```diff
@@ -488,29 +509,47 @@ func TestListSearchCaching(t *testing.T) {
...
 		}
 
 		if l := len(src.SearchCalls); l != 2 {
-			t.Errorf("Exected 2 Search calls, got %v", l)
+			t.Errorf("Executed 2 Search calls, got %v", l)
```
Comment (DavidS-ovm) [inline source_test.go:552]:
```suggestion
			t.Errorf("Expected 2 Search calls, got %v", l)
```

**Review by DavidS-ovm — COMMENTED** [review]
> it looks like in the tests e.ExecuteQuerySync(context.Background(), &req) fails for the first time for some tests

At least for TestGet/Test_Get()_caching_errors and TestListSearchCaching/non-caching_of_OTHER_errors those errors seem to be expected and should not fail the test.

For the last test case "non-caching when ignoreCache is specified", `Scope:  "error",` also seems to indicate expected errors, but I've not fully traced that down.

**Comment by tphoney** [comment]
we now check for a successful failure :P @DavidS-ovm should be good for a review

**Review by DavidS-ovm — APPROVED** [review]

**Pull request merged**